src/main/java/org/ow2/proactive/catalog/service/CatalogObjectGrantService.java

```diff
@@ -496,15 +510,24 @@ public boolean isTheUserGrantSufficientForTheCurrentTask(AuthenticatedUser user,
         // Find the catalog object id
         long catalogObjectId = getCatalogObjectId(bucketName, catalogObjectName);
         // Remove all grants that are not assigned to the current user and his group
-        grantEntities.removeIf(catalogObjectGrantEntity -> ((!catalogObjectGrantEntity.getGrantee()
-                                                                                      .equals(user.getName()) &&
-                                                             catalogObjectGrantEntity.getGrantee().equals("user")) ||
-                                                            (!user.getGroups()
-                                                                  .contains(catalogObjectGrantEntity.getGrantee()) &&
-                                                             catalogObjectGrantEntity.getGrantee().equals("group"))));
-
-        grantEntities.removeIf(catalogObjectGrantEntity -> catalogObjectGrantEntity.getCatalogObjectRevisionEntity()
-                                                                                   .getId() != catalogObjectId);
+        List<CatalogObjectGrantEntity> grantEntitiesToRemove = new LinkedList<>();
```

Comment:
why is this variable called grantEntitiesToRemove ?

Reply:
Before this section there is a function called findCatalogObjectGrantEntitiesByBucketEntityId that return all object grants inside a bucket. In grantEntitesToRemove, I put all the grants that do not belong to the current user for the current object. Then I remove them from the grant list. list.removeAll(grantEntitiesToRemove).
I tried to use a stream and a filter but i was not working properly.

Reply:
ok, then maybe you can refactor this code to a private method like filterGrantsThatDoesNotApplyToUser